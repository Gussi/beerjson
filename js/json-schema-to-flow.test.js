const convert = require('./json-schema-to-flow.js')
const rootSchema = require('../json/beer.json')

test('simple schema converted to flow', () => {
  expect(
    convert({
      definitions: {
        ObjType: {
          type: 'object',
          properties: {
            strProp: { type: 'string' },
            numberProp: { type: 'number' }
          }
        }
      }
    })
  ).toBe(
    `    export type ObjType = {
        strProp?: string,
        numberProp?: number,
    }

`
  )
})

test('enum format', () => {
  expect(
    convert({
      definitions: {
        EnumType: {
          type: 'object',
          properties: {
            enumProp: { type: 'string', enum: ['a', 'b'] }
          }
        }
      }
    })
  ).toMatchSnapshot()
})

test('root schema type', () => {
  expect(convert(rootSchema)).toMatchSnapshot()
})

test('regexp pattern should be rendered as comment', () => {
  expect(
    convert({
      definitions: {
        patternType: {
          type: 'string',
          pattern:
            '\\d{4}-\\d{2}-\\d{2}|\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}'
        }
      }
    })
  ).toMatchSnapshot()
})
