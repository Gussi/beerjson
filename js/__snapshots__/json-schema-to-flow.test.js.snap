// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enum format 1`] = `
"    type EnumType = {
        enumProp:  | \\"a\\" | \\"b\\",
    }
"
`;

exports[`root schema type 1`] = `
"     type BeerJSON = {
        version: VersionType,
        fermentables: [FermentableType],
        miscellaneous_ingredients: [MiscellaneousType],
        hop_varieties: [VarietyInformation],
        cultures: [CultureInformation],
        profiles: [WaterType],
        styles: [StyleType],
        mashes: [MashProcedureType],
        fermentations: [FermentationProcedureType],
        recipes: [RecipeType],
        equipments: [EquipmentType],
        boil: [BoilProcedureType],
        packaging: [PackagingProcedureType],
    }
"
`;
